### ISSUE TITLE
Necessary?

### ISSUE THREAD

**Issue by joefresna** [rainazakir/argos3-kilogrid#22]
If you don't do this check and you always go to DISSEMINATION, wouldn't the jump to Voter happen immediately anyway?

https://github.com/rainazakir/argos3-kilogrid/blob/b5d74efa01500872b6b49d98748f95064df7a29f/ARGoS_simulation/behaviours/agent_main.c#L504

**Comment by rainazakir** [comment]
yeah, but won't it look more organized like this? I have removed it though in 5154ac80174bab1ce86afedd7a05ed4062c6464e
